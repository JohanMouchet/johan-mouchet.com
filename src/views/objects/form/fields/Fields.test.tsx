import React from "react";
import { render } from "@testing-library/react";
import { Field, Select, Textarea } from ".";

it("Renders <Field /> without crashing", () => {
  expect(
    render(<Field type="text" name="name" value="Value" />).container.firstChild
  ).toMatchSnapshot();
});

it("Renders <Select /> without crashing", () => {
  expect(
    render(
      <Select
        name="name"
        options={[
          {
            text: "Option 1",
            value: "Option 1",
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});

it("Renders <Textarea /> without crashing", () => {
  expect(
    render(<Textarea name="name">children</Textarea>).container.firstChild
  ).toMatchSnapshot();
});
