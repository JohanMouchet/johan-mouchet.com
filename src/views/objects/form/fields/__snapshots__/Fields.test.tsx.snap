// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Field /> without crashing 1`] = `
<input
  class="o-form__field o-form__field--text"
  name="name"
  type="text"
  value="Value"
/>
`;

exports[`Renders <Select /> without crashing 1`] = `
<select
  class="o-form__field o-form__field--select"
  name="name"
>
  <option
    value="Option 1"
  >
    Option 1
  </option>
</select>
`;

exports[`Renders <Textarea /> without crashing 1`] = `
<textarea
  class="o-form__field o-form__field--textarea"
  name="name"
>
  children
</textarea>
`;
